cache/lru_cache.cc

```diff
@@ -560,7 +560,7 @@ LRUHandle* LRUCacheShard::Lookup(const Slice& key, uint32_t hash,
                                  is_in_sec_cache);
     if (secondary_handle != nullptr) {
       e = reinterpret_cast<LRUHandle*>(
-          new char[sizeof(LRUHandle) - 1 + key.size()]);
+          port::cacheline_aligned_alloc(sizeof(LRUHandle) - 1 + key.size()));
```

Comment:
This is a material space usage increase for unknown / supposed speed increase. With 16 byte key, LRUHandle is currently size ~88 bytes, which jemalloc rounds up to 96 bytes. You are rounding up to 128. Successful Lookup() always accesses the beginning (`value`) and end (`key_data`) of the LRUHandle, so it guaranteed to access two cache lines regardless of alignment efforts. Release() non-last reference only accesses `refs` so only accesses one cache line (assuming minimum alignment guaranteed by malloc). Release() last reference (LRU_Insert) modifies `next` and `m_flags` which will be in different cache lines (unless the LRUHandle starts 40 - 54 bytes offset from a cache line, which will not happen in jemalloc unless we optimize LRUHandle size down one class, and then only 1 chance in 4).

Do you have a rationale?

I'm also not sure *how* guaranteed malloc_usable_size is on the result of posix_memalign. Some searching suggests "probably."

Reply:
Thanks for the analysis - fair enough. Will switch to plain old `malloc`.